lurk-macros/src/lib.rs

```diff
@@ -118,6 +130,18 @@ fn simple_evaluate_match_arms(name: &Ident, variants: &DataEnum) -> proc_macro2:
     match_arms
 }
 
+fn has_circuit_match_arms(name: &Ident, variants: &DataEnum) -> proc_macro2::TokenStream {
```

Comment:
If the pattern becomes boring, you can have a look at the synstructure crate which might help:
https://crates.io/crates/synstructure